core/src/main/java/org/apache/iceberg/CatalogUtil.java

```diff
@@ -102,6 +106,26 @@ public static void dropTableData(FileIO io, TableMetadata metadata) {
       deleteFiles(io, manifestsToDelete);
     }
 
+    if (io instanceof SupportsBulkOperations) {
+      SupportsBulkOperations bulkFileIO = (SupportsBulkOperations) io;
+      Iterable<String> manifestPaths = Iterables.transform(manifestsToDelete, ManifestFile::path);
+      Iterable<String> prevMetadataFiles =
+          Iterables.transform(metadata.previousFiles(), TableMetadata.MetadataLogEntry::file);
+      try {
+        Iterable<String> filesToDelete =
+            Iterables.concat(
+                manifestPaths,
+                manifestListsToDelete,
+                prevMetadataFiles,
+                ImmutableList.of(metadata.metadataFileLocation()));
+        bulkFileIO.deleteFiles(filesToDelete);
```

Comment:
I feel like we should either follow what we do for non-bulk deletes and log an error message per each type or adapt the code below for non-bulk deletes to delete all files in one go.

```
if (io instanceof SupportsBulkOperations) {
  SupportsBulkOperations bulkIO = (SupportsBulkOperations) io;
  bulkDelete(bulkIO, manifests, "manifests");
  bulkDelete(bulkIO, manifestLists, "manifest lists");
  bulkDelete(bulkIO, metadataFiles, "metadata files");

} else {
  delete(io, manifests, "manifest");
  delete(io, manifestLists, "manifest list");
  delete(io, metadataFiles, "metadata file");
}
```

Or 

```
if (io instanceof SupportsBulkOperations) {
  SupportsBulkOperations bulkIO = (SupportsBulkOperations) io;
  bulkDelete(bulkIO, files);
} else {
  delete(io, files);
}
```

Reply:
Sure, I've updated so it's the 2nd way. This seems much cleaner, we delegate to the bulk deletion at the "last" layer, rather than having multiple branches. Thanks for this!